File: WandleWheelhouse/src/WandleWheelhouse.Api/DTOs/Blog/BlogArticleCardDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System;

namespace WandleWheelhouse.Api.DTOs.Blog;

#nullable enable
public class BlogArticleCardDto
{
    public Guid BlogArticleId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Excerpt { get; set; }
    public string? ImageUrl { get; set; }
    public DateTime PublicationDate { get; set; }
    public string? Slug { get; set; }

    // Author details
    public string? AuthorFullName { get; set; }
}
#nullable disable

File: WandleWheelhouse/client/src/dto/Blog/BlogArticleCardDto.ts
// src/dto/Blog/BlogArticleCardDto.ts

export interface BlogArticleCardDto {
  blogArticleId: string;
  title: string;
  excerpt?: string | null;
  imageUrl?: string | null;
  publicationDate: string; // ISO Date string
  slug?: string | null;
  authorFullName?: string | null;
  authorAvatarUrl?: string | null;
  isPublished: boolean; // <-- ADD THIS LINE
}
